app/models/link.rb

```diff
@@ -14,4 +14,12 @@ class Link < ActiveRecord::Base
       .references(:service_interactions)
       .where(service_interactions: { service_id: service })
   }
+
+  def self.get_link(params)
+    self.joins(:local_authority, :service, :interaction).find_by(
+      local_authorities: { slug: params[:local_authority_slug] },
+      services: { slug: params[:service_slug] },
+      interactions: { slug: params[:interaction_slug] }
+    )
+  end
```

Comment:
@brenetic said that when doing this as a scope it would return loads of `Link` instances instead of just one.  The reason is that scopes *always* return an array (techincally it's an `ActiveRecord::Relation` instance - which is what allows us to chain them together) - this is even if you've used a method like `.first` or `.find_by` which usually returns a single instance.  When you write it as a class method then there's none of the "magic" scope stuff happening to make sure we always return something chainable so in this case we would always get a single instance (or nil if one can't be found).